light-client/src/predicates.rs

```diff
@@ -202,6 +202,7 @@ pub trait VerificationPredicates: Send {
 /// check that their (next) validator sets hashes match.
 /// - Otherwise, ensure that the untrusted block has a greater height than
 /// the trusted block.
+#[allow(clippy::too_many_arguments)]
```

Comment:
This is not a good sign, we should look into how to refactor this. Maybe we capture a follow-up issue.

Reply:
Agreed, it's not ideal indeed. And we could perhaps improve things further by introducing typed wrappers for the untrusted/trusted blocks to prevent mixing them up. So yeah, let's follow-up on that!

Reply:
Great, think this discussion is happening in other places and shouldn't be a blocker in any form for this PR!